drivers/hmis_csv_twenty_twenty/app/models/hmis_csv_twenty_twenty/aggregated/aggregated_import_concern.rb

```diff
@@ -0,0 +1,34 @@
+###
+# Copyright 2016 - 2020 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/master/LICENSE.md
+###
+
+module HmisCsvTwentyTwenty::Aggregated::AggregatedImportConcern
+  extend ActiveSupport::Concern
+
+  included do
+    def find_matching_record
+      self.class.find_by(self.class.hud_key => self[self.class.hud_key], data_source_id: data_source_id)
```

Comment:
This should be moved to a method that can find in batches a set of hud_keys within a data source.